strutil/env.go

```diff
@@ -0,0 +1,253 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package strutil
+
+import (
+	"fmt"
+	"os"
+	"sort"
+	"strings"
+)
+
+// RawEnvironment is a slice of key=value strings.
+//
+// Using this type is required for low-level interactions, like invoking
+// execve. In other cases using Environment is both sufficient and safer.
+type RawEnvironment []string
+
+// String returns the readable, quoted representation of the raw environment.
+func (raw RawEnvironment) String() string {
+	return Quoted(raw)
+}
+
+// Environment is an unordered map of key=value strings.
+//
+// Environment can be manipulated with available methods and eventually
+// converted to RawEnvironment. This approach discourages operations that could
+// result in duplicate environment variable definitions from being constructed.
+type Environment struct {
```

Comment:
this following http.Hearer should just be:

type Enviroment map[strirng]string

Reply:
Done